File: src/rs/vision.rs
//! Interpretation of image and video values.
//!
//! Pixel transformations and still-image codecs belong to [`crate::image`];
//! video codecs and frame transformations belong to [`crate::video`]. Vision
//! is the direct Rust operation namespace for the donor's `oa::FnDetection`
//! surface; no `FnDetection` type or forwarding facade is introduced.
//! Operations are published here only when they add interpretation semantics
//! and have an operation schema, executable lowering, and correctness oracle.

mod detection;

pub use detection::{
	DetectionMetricsResult, NmsConfig, NmsResult, SegmentationMetricsResult, binary_mask_counts,
	box_iou, confusion_matrix, evaluate, evaluate_segmentation, nms,
};

File: sdk/py/binding/oa/vision.py
"""Image and video interpretation operations.
"""

from ._native import (
	DetectionMetricsResult,
	NmsResult,
	SegmentationMetricsResult,
	vision_binary_mask_counts as binary_mask_counts,
	vision_box_iou as box_iou,
	vision_confusion_matrix as confusion_matrix,
	vision_evaluate as evaluate,
	vision_evaluate_segmentation as evaluate_segmentation,
	vision_nms as nms,
)

__all__ = [
	"DetectionMetricsResult",
	"NmsResult",
	"SegmentationMetricsResult",
	"binary_mask_counts",
	"box_iou",
	"confusion_matrix",
	"evaluate",
	"evaluate_segmentation",
	"nms",
]
